### ISSUE TITLE
Cannot set DLEXT using environment variables in library/Makefile

### ISSUE THREAD

**Issue by andresag01** [ARMmbed/mbedtls#1466]
### Description
- Type: Enhancement
- Priority: Minor

----------------------------------------------------------------
## Enhancement\Feature Request

**Justification - why does the library need this feature?**  
The library/Makefile modifies overwrites the DLEXT variable value from the environment, which can be inconvenient and unexpected since it is expected to be manipulated by the user.

**Suggested enhancement**  
 We should change the assignment to DLEXT to be `?=` instead of `=`. so that we can simply build using:
```
SHARED=1 DLEXT=dylib make -C library
```

**Comment by Patater** [comment]
Consider the following.
```
ifeq ($(uname), Darwin)
DLEXT?=dylib
endif
```

**Comment by gilles-peskine-arm** [comment]
Fixed in #1499 which is now merged.

**Comment by gilles-peskine-arm** [comment]
Fixed in #1499 which is now merged.
